pkg/controller/migmigration/restore.go

```diff
@@ -285,22 +309,134 @@ func (t *Task) deleteRestores() error {
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		migapi.MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+
+	for _, gvr := range GVRs {
+		for _, ns := range t.destinationNamespaces() {
+			list, err := client.Resource(gvr).Namespace(ns).List(*listOptions)
+			if err != nil {
+				log.Trace(err)
+				return err
 			}
-			if mapping[1] != "" {
-				namespaceMapping[mapping[0]] = mapping[1]
+			for _, r := range list.Items {
+				err = client.Resource(gvr).Namespace(ns).Delete(r.GetName(), nil)
+				if err != nil {
+					// Will ignore the ones that were removed, or for some reason are not supported
+					// Assuming that main resources will be removed, such as pods and pvcs
+					if k8serror.IsMethodNotSupported(err) || k8serror.IsNotFound(err) {
+						continue
+					}
+					log.Error(err, fmt.Sprintf("Failed to request delete on: %s", gvr.String()))
+					return err
+				}
 			}
 		}
 	}
 
-	if len(namespaceMapping) != 0 {
-		restore.Spec.NamespaceMapping = namespaceMapping
+	return nil
+}
+
+func (t *Task) waitForDeleteMigrated() (bool, error) {
+	client, GVRs, err := t.getResourcesForDelete()
+	if err != nil {
+		log.Trace(err)
+		return false, err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		migapi.MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+	for _, gvr := range GVRs {
+		for _, ns := range t.destinationNamespaces() {
+			list, err := client.Resource(gvr).Namespace(ns).List(*listOptions)
+			if err != nil {
+				log.Trace(err)
+				return false, err
+			}
+			// Wait for resources with deletion timestamps
+			if len(list.Items) > 0 {
+				return false, err
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func (t *Task) deleteLabels() error {
+	client, GVRs, err := t.getResourcesForDelete()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		migapi.MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+
+	for _, gvr := range GVRs {
+		for _, ns := range t.destinationNamespaces() {
+			list, err := client.Resource(gvr).Namespace(ns).List(*listOptions)
+			if err != nil {
+				log.Trace(err)
+				return err
+			}
+			for _, r := range list.Items {
+				labels := r.GetLabels()
+				delete(labels, migapi.MigratedByLabel)
+				r.SetLabels(labels)
+				_, err = client.Resource(gvr).Namespace(ns).Update(&r, metav1.UpdateOptions{})
+				if err != nil {
+					log.Trace(err)
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (t *Task) getResourcesForDelete() (dynamic.Interface, []schema.GroupVersionResource, error) {
+	cluster, err := t.PlanResources.MigPlan.GetDestinationCluster(t.Client)
+	if err != nil {
+		log.Trace(err)
+		return nil, nil, err
+	}
+	config, err := cluster.BuildRestConfig(t.Client)
+	if err != nil {
+		log.Trace(err)
+		return nil, nil, err
+	}
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
```

Comment:
The destination client _IsA_ `compat.Client` and can be type asserted as follows:
```
discoveryClient := client.(discovery.DiscoveryInterface)
```

Reply:
Woah, didn't notice that, thanks. Will fix it now.